src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js

```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
+                  authored: moment().format('YYYY-MM-DDTHH:mm:ss[Z]'),
+                  source: patient.id,
+                  item: items,
+                },
+              },
+            }),
+          );
+        }
+        //Updating encounter
+        const cloneEncounter = { ...encounter };
+        cloneEncounter.extensionSecondaryStatus = data.nextStatus;
+        cloneEncounter.status = 'in-progress';
+        cloneEncounter.practitioner = store.getState().login.userID;
+
+        APIsArray.push(
+          FHIR('Encounter', 'doWork', {
+            functionName: 'updateEncounter',
+            functionParams: {
+              encounterId: encounter.id,
+              encounter: cloneEncounter,
+            },
+          }),
+        );
+        if (data.drugRecommendation && data.drugRecommendation.length) {
+          data.drugRecommendation.forEach((drug, drugIndex) => {
+            const medicationRequest = {};
+
+            medicationRequest['status'] = 'active';
+            medicationRequest['patient'] = patient.id;
+            medicationRequest['encounter'] = encounter.id;
+            medicationRequest['requester'] = store.getState().login.userID;
+            medicationRequest['recorder'] = store.getState().login.userID;
+            medicationRequest['note'] = drug.instructionsForTheDrug;
+            medicationRequest['routeCode'] = drug.drugRoute;
+            medicationRequest['medicationCodeableConceptCode'] = drug.drugName;
+            medicationRequest['timingCode'] = drug.intervals;
+            medicationRequest['doseQuantity'] = drug.quantity;
+            medicationRequest['methodCode'] = drug.drugForm;
+            medicationRequest['timingRepeatStart'] = moment(
+              drug.toDate,
+              'DD/MM/YYYY',
+            )
+              .subtract(drug.duration, 'days')
+              .format('YYYY-MM-DD');
+            medicationRequest['timingRepeatEnd'] = moment(
+              drug.toDate,
+              'DD/MM/YYYY',
+            ).format('YYYY-MM-DD');
+            medicationRequest['authoredOn'] = moment().format(
+              'YYYY-MM-DDTHH:mm:ss[Z]',
+            );
+
+            if (data.medicationRequest && data.medicationRequest[drugIndex]) {
+              APIsArray.push(
+                FHIR('MedicationRequest', 'doWork', {
+                  functionName: 'updateMedicationRequest',
+                  functionParams: {
+                    medicationRequest,
+                    _id: data.medicationRequest[drugIndex],
+                  },
+                }),
+              );
+            } else {
+              APIsArray.push(
+                FHIR('MedicationRequest', 'doWork', {
+                  functionName: 'createMedicationRequest',
+                  functionParams: {
+                    medicationRequest,
+                  },
+                }),
+              );
+            }
+          });
+          const res = await Promise.all[APIsArray];
```

Comment:
I think the onSubmit function needs to be in the GeneralComponents/SaveForm/index.js component.
the SaveForm need to get in the props function with all the logic of the validation and saving to a server. in the onsubmit in the  SaveForm component we will do 3 steps:
1. run the function from the parent
2. update encounter
3. move to patient tracking

when you move between tabs you dont need to implemt submit but only the function that configured in the parent.

Reply:
I'll be waiting for 14:30 stand up meeting to discuss about it so I can understand better what you meant